src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -59,10 +69,127 @@
+      }
+    }
+    return returnList;
+  }
+
+  public void checkAndFetchForDate(HistoricalPriceEntityService service, String symbols,
+      String date) {
+    int[] parsedDate = parseDateString(date);
+    int year = parsedDate[0];
+    int month = parsedDate[1];
+    int day = parsedDate[2];
+    LocalDate d = LocalDate.now().withYear(year).withMonth(month).withDayOfMonth(day);
+    LocalTime start = LocalTime.now().withHour(9).withMinute(30).withSecond(0).withNano(0);
+    LocalTime end = LocalTime.now().withHour(16).withMinute(0).withSecond(0).withNano(0);
+    for (LocalTime time = start; time.isBefore(end); time = time.plusMinutes(1)) {
+      CompositePrimaryKey key = new CompositePrimaryKey(symbols, d, time);
+      if (!service.exists(key)) {
+        List<IexHistoricalPrice> returnList = newClient.getHistoricalPricesWithDate(symbols, date);
+        for (IexHistoricalPrice price : returnList) {
+          HistoricalPriceEntity obj =
+              new HistoricalPriceEntity(price.getClose(), price.getHigh(), price.getLow(),
+                  price.getOpen(), symbols, price.getVolume(), price.getDate(),
+                  price.getMinute(), LocalTime.now());
+          service.save(obj);
+        }
+        return;
+      }
+    }
+  }
+
+  public List<IexHistoricalPrice> retrieveFromDBForDate(HistoricalPriceEntityService service,
+      String symbols, String date) {
+    List<IexHistoricalPrice> list = new ArrayList<>();
+    int[] parsedDate = parseDateString(date);
+    int year = parsedDate[0];
+    int month = parsedDate[1];
+    int day = parsedDate[2];
+    LocalDate d = LocalDate.now().withYear(year).withMonth(month).withDayOfMonth(day);
+    LocalTime start = LocalTime.now().withHour(9).withMinute(30).withSecond(0).withNano(0);
+    LocalTime end = LocalTime.now().withHour(16).withMinute(0).withSecond(0).withNano(0);
+    for (LocalTime time = start; time.isBefore(end); time = time.plusMinutes(1)) {
+      CompositePrimaryKey key = new CompositePrimaryKey(symbols, d, time);
+      Optional<HistoricalPriceEntity> temp = service.getPrices(key);
+      if (temp.isPresent()) {
+        HistoricalPriceEntity obj = temp.get();
+        IexHistoricalPrice price =
+            IexHistoricalPrice.builder().close(obj.getClose()).open(obj.getOpen())
+                .high(obj.getHigh()).low(obj.getLow()).symbol(obj.getSymbol())
+                .date(obj.getDate()).volume(obj.getVolume()).minute(obj.getMinute()).build();
+        list.add(price);
+      }
+    }
+    return list;
+  }
+
+  public int calculateRange(String rangeOfDays) {
+    if (!rangeOfDays.isEmpty()) {
+      String r = rangeOfDays.split("(?<=\\D)(?=\\d)|(?<=\\d)(?=\\D)")[0];
+      return Integer.parseInt(r);
+    }
+    return 30;
+  }
+
+  public int[] parseDateString(String date) {
+    int[] result = new int[3];
+    DateTimeFormatter formatters = DateTimeFormatter.ofPattern("yyyyMMdd");
+    LocalDate d = LocalDate.parse(date, formatters);
```

Comment:
don't store this as an array
raw arrays are bad form, and you can return a localDate instead